beacon_node/http_api/src/publish_blocks.rs

```diff
@@ -70,61 +75,101 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlock<T>>(
             .unwrap_or_else(|| Duration::from_secs(0));
 
         info!(log, "Signed block published to network via HTTP API"; "slot" => block.slot(), "publish_delay" => ?publish_delay);
-
-        let message = PubsubMessage::BeaconBlock(block);
-        crate::publish_pubsub_message(&sender, message)
-            .map_err(|_| BeaconChainError::UnableToPublish.into())
+        // Send the block, regardless of whether or not it is valid. The API
+        // specification is very clear that this is the desired behaviour.
```

Comment:
This comment is no longer meaningful (we check gossip conditions and sometimes more before publishing).

Reply:
removed here: 7bb04f5065dcc2702e08e47e7e1908abc3b6fa10